### DISCUSSION TITLE
"White space" or "whitespace"?

### DISCUSSION THREAD

**Discussion by Josh-Cena** [mdn/mdn-community#382] [Content]
I am fixing a few common spelling discrepancies, like the website/web site, webpage/web page things. "White space" is a fairly tricky one because there does not seem to be a consensus yet you would think there is. Currently we have 327 occurrences of `[Ww]hitespace`, and 405 occurrences of `[Ww]hite(.[Ss]|S)pace` in the content (including those introduced by API names; see below), so whichever we choose, there are a lot of changes to be made.

- The CSS property is called [`white-space`](https://developer.mozilla.org/en-US/docs/Web/CSS/white-space)
- The Unicode property is called [`White_Space`](https://www.unicode.org/Public/UCD/latest/ucd/PropList.txt)
- ES specs use ["White space"](https://tc39.es/ecma262/#sec-white-space)
- SVG spec uses ["White space"](https://www.w3.org/TR/SVG/text.html#WhiteSpace)
- Dictionary.com has [no entry for whitespace](https://www.dictionary.com/misspelling?term=whitespace)
- Microsoft says [white space: two words](https://learn.microsoft.com/en-us/style-guide/a-z-word-list-term-collections/w/white-space)

Seem like very strong arguments for "white space". But:

- MDN page titles and slugs generally use "Whitespace", like [How whitespace is handled by HTML, CSS, and in the DOM](https://developer.mozilla.org/en-US/docs/Web/API/Document_Object_Model/Whitespace) and even the [Whitespace glossary page](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace)
- Wikipedia uses [whitespace](https://en.wikipedia.org/wiki/Whitespace_character) for this entry
- HTML spec uses ["ASCII whitespace"](https://infra.spec.whatwg.org/#ascii-whitespace) (!)

OED is very comprehensive for this. It [says](https://www.oed.com/view/Entry/228566?redirectedFrom=white+space#eid1116831080):

> **white space**  _n._  _(a) Typography_ the blank areas of a page or other piece of printed matter, as margins, gutters between columns, etc., regarded collectively as an element of layout and design;  _(b) Computing_ (frequently as one word) blank space in electronic text produced by one or more keyed characters, as spaces, tabs, line-breaks, etc.; often _attributive_ as **whitespace character**.

...Which sounds like "use 'white space' in typography and use either in computing; attributives are usually single-word".

Seems like we have several options here:

- Bear with the inconsistency
- Use one consistently except when quoting something else like the HTML spec; there's a strong preference for "white space" in this case
- Use "white space" as noun and "whitespace" as attributive (very complex)

**Comment by estelle** [comment]
Grammarly is ok with "white space" and "whitespace" 
"white-space" is the property-name; and should not be used for anything else (Grammarly agrees).
WHATWG uses "whitespace" 216 times and "white space" zero times.
CSS Drafts use "whitespace" 48 times and "white space" 170 times, with the former (no space) being used mostly in reference to one or more characters of whitespace in code, and the latter (with space) being used in prose and in reference to handling white space overall, not as a reference to characters/entities. 

In other words, like Grammarly, I could go either way, but the CSS spec trend is interesting, something we could consider, but that is not automatable.

**Comment by bsmth** [comment]
If I see "white space" in prose, I assume we're talking about layout, more importantly with "whitespace" I can tell we're talking about characters/entities (attributively), so I agree with Estelle. It's not as simple as a search/replace to fix, unfortunately, but it's easier to disambiguate the meaning for me, at least.

**Comment by teoli2003** [comment]
So:
- "white space" when we speak about typographic or layout "empty" spaces
- "whitespace" when we speak a Unicode whitespace character
- "white-space" a CSS property name

Have I understood correctly?

**Reply by OnkarRuikar** [reply]
The ChatGPT has condensed it down to:
> Certainly! I apologize for any confusion caused. Here's a simplified explanation:
>
>The terms "whitespace" and "white space" can both be correct, depending on the context.
>
>In computer programming, "whitespace" refers to spaces, tabs, and line breaks used for formatting and organizing code. It is written as a single word, without a space in between.
>
>In a more general sense, "white space" refers to any empty or blank areas in design, typography, or layout. It is written as two separate words, with a space in between.
>
>To summarize, if you are talking about spaces used in programming, use "whitespace" as a single word. If you are referring to empty areas in design or layout, use "white space" as two separate words.

**Comment by OnkarRuikar** [comment]
If we look at numbers in `.md` files under  `en-US` dir:
`[Ww]hitespace` -  257 text + 25 code
`[Ww]hite [sS]pace` - 96 text + 1 code
`[Ww]hite[^\s][Ss]pace` - 224 text + 58 code (mostly related to syntax)

And in CSS along with `white-space` we also have `:-moz-only-whitespace` :D\
So it doesn't look strictly contextual.

I think `Bear with the inconsistency and leave it to reviewers' discretion` should be the answer. As automating this will create annoyance.
***
If we use `whitespace character` all the time instead of simply `whitespace`, then there will be less confusion among readers.

**Comment by Josh-Cena** [comment]
Thanks everyone. So do we agree on the following cases?

- "There are five whitespace characters in Unicode." (FWIW this is inconsistent with the actual naming of those characters, which is `White_Space`)
- "Whitespace characters are described in the ES standard's [White Space]() section." (Literal quoting, probably shouldn't change)
- "In JavaScript, whitespace is syntactically insignificant." (Is the source code typographical or computer?)
- "The `trim()` method removes whitespace from both ends of a string." (Is a runtime value typographical white space? Or computer whitespace?)
- "The `padding` property sets white space on the sides of the element."

**Reply by OnkarRuikar** [reply]
> * "In JavaScript, white space is syntactically insignificant." (Is the source code typographical or computer?)

For humans reading the code it's white space. For code processing tools it's whitespace characters.

**Comment by Josh-Cena** [comment]
I'm inclined to provide the following guideline:

> Use "whitespace" whenever you can substitute it with "whitespace characters" in this context; use "white space" otherwise.

Does it make sense?

**Reply by OnkarRuikar** [reply]
How about:
> Prefer term "whitespace characters" instead of "whitespace" to refer [whitespace characters](https://developer.mozilla.org/en-US/docs/Glossary/Whitespace). Otherwise, if you are referring to the general concept of empty space, you can use the term "white space."

**Reply by Josh-Cena** [reply]
My goal is to define what "general white space" means. As you see, for things like runtime string values and source code, it's not always clear.
